rasa/core/test.py

```diff
@@ -550,8 +552,20 @@ def _collect_user_uttered_predictions(
                 f" \n\n{story_dump}"
             )
     else:
+        response_selector_info = (
+            {
+                RESPONSE_SELECTOR_PROPERTY_NAME: predicted[
+                    RESPONSE_SELECTOR_PROPERTY_NAME
+                ]
+            }
+            if RESPONSE_SELECTOR_PROPERTY_NAME in predicted
+            else None
+        )
         end_to_end_user_utterance = EndToEndUserUtterance(
-            event.text, event.intent, event.entities
+            text=event.text,
+            intent=event.intent,
```

Comment:
We seem to mix labeled data here with data from the prediction. Shouldn't the retrieval intent be `intent_gold`?

Reply:
At this point, target and predicted intents are the same (there's no mismatch) so I don't think we're mixing anything. 
`intent_gold` is for labeled intent and is not a prediction. Here we need response selector prediction in order to extract it in the next step.